src/main.py

```diff
@@ -112,4 +112,10 @@ async def shutdown_event():
     app_instance = create_application()
     host = os.getenv("HOST", "0.0.0.0")
     port = int(os.getenv("PORT", "8000"))
+
+    if os.getenv("POSTGRES_USER") != "readonly_user":
+        raise ValueError("POSTGRES_USER must be set to readonly_user")
+    if os.getenv("MINIO_ACCESS_KEY") != "minio-readonly":
+        raise ValueError("MINIO_ACCESS_KEY must be set to minio-readonly")
```

Comment:
This is effectively hardcoding the user names. Not much point checking the env if they can only be one value

Reply:
That's true. My intention is to handle cases where the service might be misconfigured, rather than hardcoding the environment variable here.

Reply:
yeah, figured. The only way I can think of doing that without hardcoding the user name (and even that doesn't stop a misconfig if someone is really stupid) is querying postgres and minio to see what privs you have. I don't know how hard that is for postgres, but for minio it's really annoying - you have to list all the buckets and try to write to each one. Aaaand that doesn't stop some one updating the privs to write once the check is over.

If you don't want to hardcode the names I think you're just going to have to trust the person setting up the service. This is where I think rigorous sql parsing (later) could help, so you get two layers of protection - if one fails, hopefully the other will save you

Reply:
I think it should probably be good enough for now. Compared to other ongoing AI-related projects, this one offers a stronger protection.

Reply:
fair enough